lib/ws_orderupdates.js

```diff
@@ -0,0 +1,111 @@
+let WebSocket = require("ws");
+const Parser = require("binary-parser").Parser;
+let { CONSTANTS} = require("../config/constant");
+
+let triggers = {
+  connect: [],
+  tick: [],
+};
+
+let WSOrderUpdates = function (params) {
+  try {
+    let { clientcode, jwttoken, apikey, feedtype } = params;
+    let self = this;
+    let ws = null;
+    let headers = {
+      "x-client-code": clientcode,
+      Authorization: `Bearer ${jwttoken}`,
+      "x-api-key": apikey,
+      "x-feed-token": feedtype,
+    };
+    const url = CONSTANTS?.wsclientupdatesURL;
+    let ping_Interval = CONSTANTS?.Interval;
+    let timeStamp;
+    let stopInterval;
+    let reset;
+    let open = 1;
+
+    this.connect = function () {
+      try {
+        return new Promise((resolve, reject) => {
+          if (
+            headers?.["x-client-code"] === null ||
+            headers?.["x-feed-token"] === null ||
+            headers?.["x-api-key"] === null ||
+            headers?.Authorization === null
+          ) {
+            return "client_code or jwt_token or api_key or feed_token is missing";
+          }
+          ws = new WebSocket(url, { headers });
+
+          ws.onopen = function onOpen(evt) {
+            reset = setInterval(function () {
+              ws.send("ping");
+            }, ping_Interval);
+            resolve();
+          };
+
+          ws.onmessage = function (evt) {
+            let result = evt.data;
+            timeStamp = Math.floor(Date.now() / 1000);
+            trigger("tick", [result]);
+            resolve(result);
+          };
+
+          stopInterval = setInterval(function () {
+            let currentTimeStamp = Math.floor(Date.now() / 1000);
+            let lastMessageTimeStamp = currentTimeStamp - timeStamp;
+            if (lastMessageTimeStamp > 20) {
```

Comment:
provide comment above the line  if (lastMessageTimeStamp > 20) {

Reply:
Comment has been added